src/xdg.rs

```diff
@@ -262,6 +262,63 @@ impl IconLoader {
         Self { icons }
     }
 
+    fn premultiply_generic(data: &mut [u8]) {
+        // TODO: change to array_chunks_mut() once that is stabilised.
+        for chunk in data.chunks_exact_mut(4) {
+            if let [r, g, b, a] = chunk {
```

Comment:
Correct, I had an `else { std::hint::unreachable_unchecked() }` before, but it was unnecessary, the compiler is smart enough to know it will only ever emit four values in this slice.